doc/releases/reef.rst

```diff
@@ -0,0 +1,266 @@
+  the address of a module's RADOS client, but not the name of the module.
+* The ``perf dump`` and ``perf schema`` commands have been deprecated in 
+  favor of the new ``counter dump`` and ``counter schema`` commands. These new
+  commands add support for labeled perf counters and also emit existing
+  unlabeled perf counters. Some unlabeled perf counters became labeled in this
+  release, and more will be labeled in future releases; such converted perf
+  counters are no longer emitted by the ``perf dump`` and ``perf schema``
+  commands.
+* core: Cache tiering is now deprecated.
+* The SPDK back end for BlueStore can now connect to an NVMeoF target. This
+  is not an officially supported feature.
+
+RBD
+~~~
+
+* The semantics of compare-and-write C++ API (`Image::compare_and_write` and
+  `Image::aio_compare_and_write` methods) now match those of C API. Both
+  compare and write steps operate only on len bytes even if the buffers
+  associated with them are larger. The previous behavior of comparing up to the
+  size of the compare buffer was prone to subtle breakage upon straddling a
+  stripe unit boundary.
+* RBD: The ``compare-and-write`` operation is no longer limited to 512-byte 
+  sectors. Assuming proper alignment, it now allows operating on stripe units
+  (4MB by default).
+* RBD: There is a new ``rbd_aio_compare_and_writev`` API method that supports 
+  scatter/gather on compare buffers as well as on write buffers. This
+  complements the existing ``rbd_aio_readv`` and ``rbd_aio_writev`` methods.
+* RBD: The ``rbd device unmap`` command now has a ``--namespace`` option.
+  Support for namespaces was added to RBD in Nautilus 14.2.0, and since then it
+  has been possible to map and unmap images in namespaces using the
+  ``image-spec`` syntax. However, the corresponding option available in most
+  other commands was missing.
+* RBD: All rbd-mirror daemon perf counters have become labeled and are now
+  emitted only by the new ``counter dump`` and ``counter schema`` commands. As
+  part of the conversion, many were also renamed in order to better
+  disambiguate journal-based and snapshot-based mirroring.
+* RBD: The list-watchers C++ API (`Image::list_watchers`) now clears the passed
+  `std::list` before appending to it. This aligns with the semantics of the C
+  API (``rbd_watchers_list``).
+* RBD: Trailing newline in passphrase files (for example: the
+  ``<passphrase-file>`` argument of the ``rbd encryption format`` command and
+  the ``--encryption-passphrase-file`` option of other commands) is no longer
+  stripped.
+* RBD: Support for layered client-side encryption has been added. It is now
+  possible to encrypt cloned images with a distinct encryption format and
+  passphrase, differing from that of the parent image and from that of every
+  other cloned image. The efficient copy-on-write semantics intrinsic to
+  unformatted (regular) cloned images have been retained.
+
+RGW
+~~~
+
```

Comment:
sorry, we'd forgotten to document the most important rgw feature!

* Bucket resharding is now supported for multi-site configurations. This feature will be enabled by default for new deployments. Existing deployments must enable the ``resharding`` feature manually after all zones have upgraded. See https://docs.ceph.com/en/reef/radosgw/multisite/#zone-features for details.

Reply:
Added at the top of the list of RGW changes so that it will appear in the most prominent location.